File: Plugins/Aspose_Tasks_Java_for_Ruby/lib/asposetasksjava/ResourceAssignments/assignmentbudget.rb
module Asposetasksjava
  module AssignmentBudget
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new(data_dir + 'test_tasks.mpp')

        resource_assignments = project.getResourceAssignments().toList()

        # Parse through all the collected resource assignments
        i = 0
        while i < resource_assignments.size()
          ra = resource_assignments.get(i)
          puts "Buget Cost: " + ra.get(Rjb::import('com.aspose.tasks.Asn').BUDGET_COST).toString()
          puts "Buget Work: " + ra.get(Rjb::import('com.aspose.tasks.Asn').BUDGET_WORK).toString()
          puts "--------------------------------------------------------"
          i +=1
        end    
    end
  end
end

File: Plugins/Aspose_Tasks_Java_for_PHP/src/aspose/tasks/WorkingWithResourceAssignments/AssignmentBudget.php
<?php
namespace Aspose\Tasks\WorkingWithResourceAssignments;

use com\aspose\tasks\Project as Project;
use com\aspose\tasks\Asn as Asn;
class AssignmentBudget{

    public static function run($dataDir=null)
    {

        # Instantiate project object
        $project = new Project($dataDir . 'test_tasks.mpp');

        $resource_assignments = $project->getResourceAssignments()->toList();

        # Parse through all the collected resource assignments
        $asn=new Asn();
        $i = 0;
        while ($i < sizeof($resource_assignments)){
            $ra = $resource_assignments->get($i);
          print "Buget Cost: " . (string)$ra -> get($asn -> BUDGET_COST).PHP_EOL;// . toString()
          print "Buget Work: " . (string)$ra -> get($asn -> BUDGET_WORK).PHP_EOL;// . toString()
          print "--------------------------------------------------------" . PHP_EOL;
          $i += 1;
        }
    }
}
?>